File: analysis/load.py
import json
import os
from typing import Any

repos: dict[str, dict[str, Any]] = {}

for repo in os.listdir("repositories"):
    issues = repos[repo] = {}
    for issue in os.listdir(os.path.join("repositories", repo, "issues")):
        with open(os.path.join("repositories", repo, "issues", issue), "rb") as file:
            issues[issue] = json.load(file)
    for pull in os.listdir(os.path.join("repositories", repo, "pulls")):
        with open(os.path.join("repositories", repo, "pulls", pull), "rb") as file:
            issues[pull] = json.load(file)

for repo, issues in sorted(repos.items(), key=lambda x: x[0]):
    print(f"{repo}: {len(issues)}")

File: analysis/load.cpp
#include <filesystem>
#include <fstream>
#include <iostream>
#include <map>
#include <nlohmann/json.hpp>
#include <string>

using json = nlohmann::json;

int main()
{
    std::map<std::string, std::map<std::string, json>> repos;

    for (const auto& repo : std::filesystem::directory_iterator("repositories")) {
        std::map<std::string, json> issues;
        for (const auto& issue : std::filesystem::directory_iterator(repo.path() / "issues")) {
            std::ifstream file(issue.path());
            issues[issue.path().filename()] = json::parse(file);
        }
        for (const auto& pull : std::filesystem::directory_iterator(repo.path() / "pulls")) {
            std::ifstream file(pull.path());
            issues[pull.path().filename()] = json::parse(file);
        }
        repos[repo.path().filename()] = issues;
    }

    for (const auto& [repo, issues] : repos) {
        std::cout << repo << ": " << issues.size() << "\n";
    }

    return 0;
}
